packages/admin/src/pages/build-application/[applicationId]/[sectionId]/[questionId]/edit/question-content/getServerSideProps.tsx

```diff
@@ -0,0 +1,83 @@
+import { GetServerSidePropsContext } from 'next';
+import { getApplicationFormSummary } from '../../../../../../../services/ApplicationService';
+import {
+  getQuestion,
+  patchQuestion,
+} from '../../../../../../../services/QuestionService';
+import { getSessionIdFromCookies } from '../../../../../../../utils/session';
+import QuestionPageGetServerSideProps from '../../../../../../../utils/QuestionPageGetServerSideProps';
+import { NextRedirect } from '../../../../../../../utils/QuestionPageGetServerSidePropsTypes';
+
+type RequestBody = {
+  fieldTitle: string;
+  hintText: string;
+  optional: string;
+  maxWords?: string;
+};
+
+const getServerSideProps = (context: GetServerSidePropsContext) => {
+  const { applicationId, sectionId, questionId } = context.params as Record<
+    string,
+    string
+  >;
+  const { backTo } = context.query as Record<string, string>;
+
+  function onSuccessRedirectHref() {
+    return backTo === 'dashboard'
+      ? `/build-application/${applicationId}/dashboard`
+      : `/build-application/${applicationId}/${sectionId}`;
+  }
+
+  async function fetchPageData(jwt: string) {
+    const applicationFormSummary = await getApplicationFormSummary(
+      applicationId,
+      jwt
+    );
+
+    if (applicationFormSummary.applicationStatus === 'PUBLISHED') {
+      return {
+        redirect: {
+          destination: `/build-application/${applicationId}/${sectionId}/${questionId}/preview`,
+          statusCode: 302,
```

Comment:
```suggestion
          permanent: false,
```

> Why does Next.js use 307 and 308? Traditionally a 302 was used for a temporary redirect, and a 301 for a permanent redirect, but many browsers changed the request method of the redirect to GET, regardless of the original method. For example, if the browser made a request to POST /v1/users which returned status code 302 with location /v2/users, the subsequent request might be GET /v2/users instead of the expected POST /v2/users. Next.js uses the 307 temporary redirect, and 308 permanent redirect status codes to explicitly preserve the request method used.

Reply:
nvm, dom explained that we actually need the method overwrite here